tests/test_default_impls.rs

```diff
@@ -0,0 +1,156 @@
+#![allow(non_snake_case)]
```

Comment:
Heh, I see what you mean about these tests being a bit painful!

They're very coupled to the implementation, which is going to hurt whenever things get refactored.

How about instead we go a bit further in this PR and implement a default `__repr__` for the `#[pyclass]` enums. Then the tests for the overridable slots can instead be that the default `__repr__` works and a second test to confirm it's also overrideable.

... and similar for the overrideable methods, however there isn't a method we want to default right now which springs to mind.

Reply:
> however there isn't a method we want to default right now which springs to mind.

Maybe we don't support ordinary methods in this PR? We can always add a new trait later if that's needed.

Reply:
Sounds reasonable to me!